ketama/ketama.go

```diff
@@ -36,6 +38,8 @@ type Continuum struct {
 	numpoints int
 	modtime   time.Time
 	array     mcsArray
+	newHash   func() hash.Hash
```

Comment:
`libmemcached` has two separate configurable hashes for the ketama continuum: The `mc key -> number` function (`MEMCACHED_BEHAVIOR_HASH` in libmemcached) and the `server -> number` function (`MEMCACHED_BEHAVIOR_KETAMA_HASH` in libmemcached). 

This appears to use the same function for both, which I think is fine, but useful to call out as there might be some awful future where we need to remember and implement two different configurable hash functions here.